components/omega/src/ocn/TendencyTerms.cpp

```diff
@@ -12,9 +12,197 @@
+
+} // end erase
+
+//------------------------------------------------------------------------------
+// Get default tendencies
+Tendencies *Tendencies::getDefault() {
+
+   return Tendencies::DefaultTendencies;
+
+} // end get default
+
+//------------------------------------------------------------------------------
+// Get state by name
+Tendencies *Tendencies::get(const std::string Name ///< [in] Name of tendencies
+) {
+
+   auto it = AllTendencies.find(Name);
+
+    if (it != AllTendencies.end()) {
+      return &(it->second);
+    } else {
+      LOG_ERROR(
+          "Tendencies::get: Attempt to retrieve non-existent tendencies:");
+      LOG_ERROR("{} has not been defined or has been removed", Name);
+      return nullptr;
+    }
+
+} // end get state
+
+//------------------------------------------------------------------------------
+// Construct a new group of tendencies
+Tendencies::Tendencies(const std::string &Name, ///< [in] Name for tendencies
+                       const HorzMesh *Mesh,    ///< [in] Horizontal mesh
+                       int NVertLevels, ///< [in] Number of vertical levels
+                       Config *Options  ///< [in] Configuration options
+                       )
+    : ThicknessFluxDiv(Mesh, Options), PotientialVortHAdv(Mesh, Options),
+      KEGrad(Mesh, Options), SHHGrad(Mesh, Options),
+      VelocityDiffusion(Mesh, Options), VelocityHyperDiff(Mesh, Options) {
+
+   // Tendency arrays
+   LayerThicknessTend =
+       Array2DReal("LayerThicknessTend", Mesh->NCellsSize, NVertLevels);
+   NormalVelocityTend =
+       Array2DReal("NormalVelocityTend", Mesh->NEdgesSize, NVertLevels);
+
+   //
+   NCellsOwned = Mesh->NCellsOwned;
+   NEdgesOwned = Mesh->NEdgesOwned;
+   NChunks     = NVertLevels / VecLength;
+
+   // Associate this instance with a name
+   AllTendencies.emplace(Name, *this);
+
+} // end constructor
+
+//------------------------------------------------------------------------------
+// Compute tendencies for layer thickness equation
+// TODO Add AuxilaryState as argument
+void Tendencies::computeThicknessTendencies(
+    OceanState *State ///< [in] State variables
+) {
+
+   OMEGA_SCOPE(o_LayerThicknessTend, LayerThicknessTend);
+   OMEGA_SCOPE(o_NCellsOwned, NCellsOwned);
+   OMEGA_SCOPE(o_NChunks, NChunks);
+   OMEGA_SCOPE(o_ThicknessFluxDiv, ThicknessFluxDiv);
+
+   // Compute thickness flux divergence
+   Array2DReal ThickFluxEdge;
+   parallelFor(
+       {o_NCellsOwned, o_NChunks}, KOKKOS_LAMBDA(int ICell, int KChunk) {
+          o_ThicknessFluxDiv(o_LayerThicknessTend, ICell, KChunk, ThickFluxEdge);
+       });
+
+} // end thickness tehndency compute
+
+//------------------------------------------------------------------------------
+// Compute tendencies for normal velocity equation
+// TODO Add AuxilaryState as argument
+void Tendencies::computeVelocityTendencies(
+    OceanState *State ///< [in] State variables
+) {
+
+   OMEGA_SCOPE(o_NormalVelocityTend, NormalVelocityTend);
```

Comment:
See https://github.com/E3SM-Project/Omega/pull/92#discussion_r1670441698. We need to pick a convention and stick to it.

Reply:
Good point, I went to using the `Loc` prefix you used in #112.